python-sdk/src/astro/databases/mssql.py

```diff
@@ -0,0 +1,382 @@
+
+    def create_schema_if_needed(self, schema: str | None) -> None:
+        """
+        This function checks if the expected schema exists in the database. If the schema does not exist,
+        it will attempt to create it.
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
+        """
+        if schema and not self.schema_exists(schema):
+            statement = self._create_schema_statement.format(schema)
+            self.run_sql(statement, autocommit=True)
+
+    def create_table_from_select_statement(
+        self,
+        statement: str,
+        target_table: BaseTable,
+        parameters: dict | None = None,
+    ) -> None:
+        """
+        Export the result rows of a query statement into another table.
+        :param statement: SQL query statement
+        :param target_table: Destination table where results will be recorded.
+        :param parameters: (Optional) parameters to be used to render the SQL query
+        """
+
+        statement = self._create_table_statement.format(
+            target_table.metadata.schema,
+            target_table.name,
+            target_table.metadata.database,
+            statement,
+        )
+
+        self.run_sql(statement, parameters, autocommit=True)
+
+    def drop_table(self, table: BaseTable) -> None:
+        """
+        Delete a SQL table, if it exists.
+
+        :param table: The table to be deleted.
+        """
+        statement = self._drop_table_statement.format(self.get_table_qualified_name(table))
+        self.run_sql(statement, autocommit=True)
+
+    def fetch_all_rows(self, table: BaseTable, row_limit: int = -1) -> list:
+        """
+        Fetches all rows for a table and returns as a list. This is needed because some
+        databases have different cursors that require different methods to fetch rows
+
+        :param row_limit: Limit the number of rows returned, by default return all rows.
+        :param table: The table metadata needed to fetch the rows
+        :return: a list of rows
+        """
+        statement = f"SELECT * FROM {self.get_table_qualified_name(table)}"  # skipcq: BAN-B608
+        if row_limit > -1:
+            statement = f"SELECT TOP {row_limit} * FROM {self.get_table_qualified_name(table)}"
+        response = self.run_sql(statement)
+        return response.fetchall()  # type: ignore
+
+    # Require skipcq because method overriding we need param chunk_size
+    def load_pandas_dataframe_to_table(
+        self,
+        source_dataframe: pd.DataFrame,
+        target_table: BaseTable,
+        if_exists: LoadExistStrategy = "replace",
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+    ) -> None:  # skipcq PYL-W0613
+        """
+        Create a table with the dataframe's contents.
+        If the table already exists, append or replace the content, depending on the value of `if_exists`.
+        :param source_dataframe: Local or remote filepath
```

Comment:
```suggestion
        If the table already exists, append or replace the content, depending on the value of `if_exists`.

        :param source_dataframe: Local or remote filepath
```

Reply:
and same everywhere else. Check whitespace error code in https://www.pydocstyle.org/en/stable/error_codes.html or https://google.github.io/styleguide/pyguide.html#381-docstrings

Reply:
fixed